src/main/java/com/intuit/apl/engine/Context.java

```diff
@@ -250,6 +251,41 @@ public boolean containsAllIgnoreCase(ArrayList<String> source, String target) {
     return true;
   }
 
+  public boolean verifyRequest(){
+    return verifyRequest(subject) && verifyRequest(resource) && verifyRequest(action) && verifyRequest(environment);
+  }
+
+  private boolean verifyRequest(Map<String, String> map){
+    if(CollectionUtils.isEmpty(map)){
+      return true;
+    }
+
+    Map<String, String> verifyMap = getPrefixMap(map, "verify_");
+
+    for(Map.Entry<String, String> entry : verifyMap.entrySet()){
+      String verifyKey = entry.getKey();
+      if(!map.containsKey(verifyKey)){
+        return false;
+      }
+      String value = map.get(verifyKey);
+      String regex = entry.getValue();
+      ArrayList<String> regexList = (ArrayList<String>) Arrays.asList(regex);
+      if(!containsAny(regexList, value) && !matchRegex(entry.getValue(), map.get(verifyKey))){
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private Map<String, String> getPrefixMap(Map<String, String> map, String prefix){
+    return map.entrySet().stream().filter(entry -> entry.getKey().startsWith(prefix))
+            .collect(Collectors.toMap(entry -> entry.getKey().replaceFirst(prefix, ""), Map.Entry::getValue));
+  }
+
+  private boolean matchRegex(String regex, String value){
+    return Pattern.compile(regex).matcher(value).matches();
```

Comment:
We also need to check if it is a valid regex pattern. If not we need to handle that gracefully.